internal/services/healthcare/healthcare_fhir_resource.go

```diff
@@ -0,0 +1,635 @@
+		Read:   resourceHealthcareApisFhirServiceRead,
+		Update: resourceHealthcareApisFhirServiceUpdate,
+		Delete: resourceHealthcareApisFhirServiceDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.FhirServiceID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.FhirServiceName(),
+			},
+
+			"resource_group_name": commonschema.ResourceGroupName(),
+
+			"workspace_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.WorkspaceID,
+			},
+
+			"location": commonschema.Location(),
+
+			"kind": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  string(healthcareapis.KindFhirR4),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(healthcareapis.KindFhirR4),
+					string(healthcareapis.KindFhirStu3),
+				}, false),
+			},
+
+			"access_policy_object_ids": {
+				Type:     pluginsdk.TypeSet,
+				Optional: true,
+				Elem: &pluginsdk.Schema{
+					Type:         pluginsdk.TypeString,
+					ValidateFunc: validation.IsUUID,
+				},
+			},
+
+			"authentication": {
+				Type:     pluginsdk.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"authority": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+						"audience": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+						},
+						"smart_proxy_enabled": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"identity": commonschema.SystemAssignedIdentityOptional(),
+
+			// can't use the registry ID due to the ID cannot be obtained when setting the property in state file
+			"acr_login_servers": {
+				Type:     pluginsdk.TypeSet,
+				Optional: true,
+				Elem: &pluginsdk.Schema{
+					Type:         pluginsdk.TypeString,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+			},
+
+			"cors_configuration": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"allowed_origins": {
+							Type:     pluginsdk.TypeSet,
+							Optional: true,
+							MaxItems: 64,
+							Elem: &pluginsdk.Schema{
+								Type:         pluginsdk.TypeString,
+								ValidateFunc: validation.StringIsNotEmpty,
+							},
+						},
+						"allowed_headers": {
```

Comment:
could we be consistent with spacing here?
```suggestion
							},
						},

						"allowed_headers": {
```